### PR TITLE
Correcting Xcode build errors

### PR DESCRIPTION
**Pull request by rawbengal**
These two changes correct some build errors when including OLA headers in other programs.

### CODE REVIEW

**Comment by peternewman** [comment]
Thanks @rawbengal , we need to cherry-pick this into 0.9 when it's in.

File: include/ola/client/ClientArgs.h [hunk 1]
```diff
@@ -90,9 +90,9 @@ struct SendDMXArgs {
   /**
    * @brief Create a new SendDMXArgs object
    */
-  explicit SendDMXArgs(GeneralSetCallback *callback)
+  explicit SendDMXArgs(GeneralSetCallback *_callback)
```
Comment (peternewman) [inline include/ola/client/ClientArgs.h]: We normally name the field m_, would you mind doing that please and dropping the leading underscores from the constructor arguments.

File: include/ola/messaging/StringMessageBuilder.h [hunk 2]
```diff
@@ -38,7 +38,7 @@ class StringMessageBuilder: public FieldDescriptorVisitor {
     explicit StringMessageBuilder(const std::vector<std::string> &input)
         : m_input(input) {
     }
-    ~StringMessageBuilderVisitor() {}
+    ~StringMessageBuilder() {}
```
Comment (peternewman) [inline include/ola/messaging/StringMessageBuilder.h]: Good catch, was it Clang reported this (and if so which version), or Xcode itself?

File: include/ola/client/ClientArgs.h (hunk 1 above)
Comment (nomis52) [inline include/ola/client/ClientArgs.h]: Not for option structs. We shouldn't force the clients to do:

options.m_priority = X;

but rather:

options.priority = X;

I don't mind what the constructor args are called. I would have gone with callback_arg but _callback works as well.

**Comment by nomis52** [comment]
Can you please close this PR and open another one targeting the 0.9. This is a bugfix so it belongs there. It'll get merged back into the mainline later.

File: include/ola/client/ClientArgs.h (hunk 1 above)
Comment (peternewman) [inline include/ola/client/ClientArgs.h]: Ah okay.

File: include/ola/messaging/StringMessageBuilder.h (hunk 2 above)
Comment (rawbengal) [inline include/ola/messaging/StringMessageBuilder.h]: Xcode.  I don't recall seeing any warnings when compiling the library...  just when using the header in an Xcode project.

**Comment by rawbengal** [comment]
Copy that.  Closing this PR and will target 0.9.   It might take me a bit to figure out how to do move the change to 0.9.  Git is rarely obvious to me.

**Comment by peternewman** [comment]
@rawbengal look at git cherry-pick, if you switch to the 0.9 branch, you can then fetch, rather than pull your master an then cherry pick the commits from it.

**Comment by nomis52** [comment]
When you open the PR, you can select which branch you target. See  https://help.github.com/articles/using-pull-requests/#changing-the-branch-range-and-destination-repository  for screenshot

**Comment by peternewman** [comment]
Unfortunately he's too late for that @nomis52 I tried another comparison against 0.9 and he's initially pulled from our master, so merging in like that ends up merging all our changes from master into 0.9. I'd suggest we re-open this, merge given we seem to be happy and cherry-pick ourselves.

**Comment by nomis52** [comment]
https://github.com/OpenLightingProject/ola/pull/840 is correct

**Pull request closed without merging**